frontend/packages/ceph-storage-plugin/src/components/dashboard-page/storage-dashboard/status-card/status-card.tsx

```diff
@@ -0,0 +1,121 @@
+import * as React from 'react';
+import { Gallery, GalleryItem } from '@patternfly/react-core';
+import {
+  DashboardCardHeader,
+  DashboardCard,
+  DashboardCardTitle,
+  DashboardCardBody,
+} from '@console/internal/components/dashboard/dashboard-card';
+import { AlertsBody } from '@console/internal/components/dashboard/status-card/status-body';
+import { HealthBody } from '@console/internal/components/dashboard/status-card/health-body';
+import { HealthItem } from '@console/internal/components/dashboard/status-card/health-item';
+import {
+  withDashboardResources,
+  DashboardItemProps,
+} from '@console/internal/components/dashboards-page/with-dashboard-resources';
+import { AlertItem } from '@console/internal/components/dashboard/status-card/alert-item';
+import { getAlerts } from '@console/internal/components/dashboard/health-card';
+import { ALERTS_KEY } from '@console/internal/actions/dashboards';
+import { PrometheusResponse } from '@console/internal/components/graphs';
+import { PrometheusRulesResponse, alertURL } from '@console/internal/components/monitoring';
+import {
+  STORAGE_HEALTH_QUERIES,
+  DATA_RESILIENCY_QUERIES,
+  StorageDashboardQuery,
+} from '../../../../constants/queries';
+import { filterCephAlerts } from '../../../../selectors';
+import { cephClusterResource } from '../../../../constants/resources';
+import { getCephHealthState, getDataResiliencyState } from './utils';
+
+export const StatusCard = withDashboardResources(
+  ({
+    watchK8sResource,
+    stopWatchK8sResource,
+    watchPrometheus,
+    stopWatchPrometheusQuery,
+    prometheusResults,
+  }: DashboardItemProps) => {
+    React.useEffect(() => {
+      watchK8sResource(cephClusterResource);
+      watchPrometheus(STORAGE_HEALTH_QUERIES[StorageDashboardQuery.CEPH_STATUS_QUERY]);
+      watchPrometheus(DATA_RESILIENCY_QUERIES[StorageDashboardQuery.RESILIENCY_PROGRESS]);
+
+      return () => {
+        stopWatchK8sResource(cephClusterResource);
+        stopWatchPrometheusQuery(STORAGE_HEALTH_QUERIES[StorageDashboardQuery.CEPH_STATUS_QUERY]);
+        stopWatchPrometheusQuery(
+          DATA_RESILIENCY_QUERIES[StorageDashboardQuery.RESILIENCY_PROGRESS],
+        );
+      };
+    }, [watchK8sResource, stopWatchK8sResource, watchPrometheus, stopWatchPrometheusQuery]);
+
+    const cephStatusQuery = prometheusResults.getIn([
+      STORAGE_HEALTH_QUERIES[StorageDashboardQuery.CEPH_STATUS_QUERY],
+      'data',
+    ]) as PrometheusResponse;
+    const cephStatusQueryError = prometheusResults.getIn([
+      STORAGE_HEALTH_QUERIES[StorageDashboardQuery.CEPH_STATUS_QUERY],
+      'loadError',
+    ]);
+    const resiliencyProgressResult = prometheusResults.getIn([
+      DATA_RESILIENCY_QUERIES[StorageDashboardQuery.RESILIENCY_PROGRESS],
+      'data',
+    ]) as PrometheusResponse;
+    const resiliencyProgressError = prometheusResults.getIn([
+      DATA_RESILIENCY_QUERIES[StorageDashboardQuery.RESILIENCY_PROGRESS],
+      'loadError',
+    ]);
```

Comment:
Please add data type here as well similar to other prometheus response.

Reply:
It comes and empty string and a TypeError object.
I do not see it defined anywhere, we can leave it!

Reply:
ack